gtda/images/preprocessing.py

```diff
@@ -108,17 +108,95 @@ def transform(self, X, y=None):
         -------
         Xt : ndarray, shape (n_samples, n_pixels_x, n_pixels_y [, n_pixels_z])
             Transformed collection of images. Each entry along axis 0 is a
-            2D or 3D binary images.
+            2D or 3D binary image.
         """
 
         check_is_fitted(self)
-        X = check_array(X,  ensure_2d=False, allow_nd=True)
+        Xt = check_array(X,  ensure_2d=False, allow_nd=True, copy=True)
 
-        Xt = Parallel(n_jobs=self.n_jobs)(delayed(self._binarize)(X[s])
-            for s in gen_even_slices(X.shape[0], effective_n_jobs(self.n_jobs)))
+        Xt = Parallel(n_jobs=self.n_jobs)(delayed(
+            self._binarize)(X[s])
+            for s in gen_even_slices(X.shape[0],
+                                     effective_n_jobs(self.n_jobs)))
         Xt = np.concatenate(Xt)
 
         if self.n_dimensions_ == 2:
             Xt = Xt.reshape((*X.shape))
 
         return Xt
+
+
+@adapt_fit_transform_docs
+class Inverter(BaseEstimator, TransformerMixin):
+    """Transformer returning a collection of binary images from an input
+    collection of 2D or 3D grayscale images.
+
+    Parameters
+    ----------
+    n_jobs : int or None, optional, default: ``None``
+        The number of jobs to use for the computation. ``None`` means 1 unless
+        in a :obj:`joblib.parallel_backend` context. ``-1`` means using all
+        processors.
+
+    """
+
+    def __init__(self, n_jobs=None):
+        self.n_jobs = n_jobs
+
+    def fit(self, X, y=None):
+        """Do nothing and return the estimator unchanged.
+
+        This method is here to implement the usual scikit-learn API and hence
+        work in pipelines.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_samples, n_pixels_x, n_pixels_y [, n_pixels_z])
+            Input data. Each entry along axis 0 is interpreted as a 2D or 3D
+            binary image.
+
+        y : None
+            There is no need of a target in a transformer, yet the pipeline API
+            requires this parameter.
+
+        Returns
+        -------
+        self : object
+
+        """
+        X = check_array(X, ensure_2d=False, allow_nd=True)
+
+        self._is_fitted = True
+        return self
+
+    def transform(self, X, y=None):
+        """For each collection of binary images, calculate the corresponding
+        collection of binary images based on their logical negation.
```

Comment:
In general, I am not a fan of `For each collection of binary images,...`, since, in a call to transform, you  are dealing with *one* collection of images.
Perhaps something like
```
"""For each image in the collection `X`, calculate its negation. Return the collection of negated images."""
```

Reply:
Nice, thanks!